matching-core/src/main/java/com/graphhopper/matching/MapMatching.java

```diff
@@ -191,122 +196,146 @@ public MatchResult doWork(List<GPXEntry> gpxList) {
-                } else {
-                	computeTransitionProbabilities(prevTimeStep, timeStep, probabilities, queryGraph);
-                	viterbi.nextStep(timeStep.observation, timeStep.candidates, timeStep.emissionLogProbabilities,
-                			timeStep.transitionLogProbabilities, timeStep.roadPaths);
+            computeEmissionProbabilities(timeStep, probabilities);
+
+            if (prevTimeStep == null) {
+                viterbi.startWithInitialObservation(timeStep.observation, timeStep.candidates,
+                        timeStep.emissionLogProbabilities);
+            } else {
+                computeTransitionProbabilities(prevTimeStep, timeStep, probabilities, queryGraph);
+                viterbi.nextStep(timeStep.observation, timeStep.candidates,
+                        timeStep.emissionLogProbabilities, timeStep.transitionLogProbabilities,
+                        timeStep.roadPaths);
+            }
+            if (viterbi.isBroken()) {
+                String likelyReasonStr = "";
+                if (prevTimeStep != null) {
+                     GPXEntry prevGPXE = prevTimeStep.observation;
+                     GPXEntry gpxe = timeStep.observation;
+                     double dist = distanceCalc.calcDist(prevGPXE.lat, prevGPXE.lon,
+                             gpxe.lat, gpxe.lon);
+                     if (dist > 2000) {
+                         likelyReasonStr = "Too long distance to previous measurement? "
+                                 + Math.round(dist) + "m, ";
+                     }
                 }
-                if (viterbi.isBroken()) {
-                    String likelyReasonStr = "";
-                    if (prevTimeStep != null) {
-                    	 GPXEntry prevGPXE = prevTimeStep.observation;
-                         GPXEntry gpxe = timeStep.observation;
-                         double dist = distanceCalc.calcDist(prevGPXE.lat, prevGPXE.lon, gpxe.lat, gpxe.lon);
-                         if (dist > 2000) {
-                             likelyReasonStr = "Too long distance to previous measurement? " + Math.round(dist) + "m, ";
-                         }
-                    }
 
-                    throw new RuntimeException("Sequence is broken for submitted track at time step " + timeStepCounter 
-                    		+ " (" + gpxList.size() + " points). "
-                            + likelyReasonStr
-                            + "observation:" + timeStep.observation + ", candidates: " + getSnappedCandidates(timeStep.candidates));
-                }
-                
-                timeStepCounter++;
-                prevTimeStep = timeStep;
+                throw new RuntimeException("Sequence is broken for submitted track at time step "
+                        + timeStepCounter
+                        + " (" + gpxList.size() + " points). "
+                        + likelyReasonStr
```

Comment:
The line with "throw new RuntimeException(..." was longer than 100 characters. However, I removed line breaks of the following lines to reduce the number of lines.